### PR TITLE
Support pure-capability RISC-V compiler-rt builtins

### PR DESCRIPTION
**Pull request by jrtc27**
Supersedes #533 and #545.

### CODE REVIEW

**Review by heshamelmatary — APPROVED** [review]

**Review by arichardson — DISMISSED** [review]
LGTM but marking as request changes so we don't merge this into dev until the upstream merge has completed.

**Comment by jrtc27** [comment]
> LGTM but marking as request changes so we don't merge this into dev until the upstream merge has completed.

Nice idea but doesn't actually block it for admins 😂

**Comment by arichardson** [comment]
> > LGTM but marking as request changes so we don't merge this into dev until the upstream merge has completed.
> 
> Nice idea but doesn't actually block it for admins 😂

I thought it changed the merge button to be red so we don't accidentally do it. But it doesn't appear to have changed anything for me...

**Pull request merged**
